src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/NetCorePackageReferenceProject.cs

```diff
@@ -214,40 +217,111 @@ private void IgnoreUnsupportProjectReference(PackageSpec project)
 
         #region NuGetProject
 
-        public override Task<IEnumerable<PackageReference>> GetInstalledPackagesAsync(CancellationToken token)
+        public async override Task<IEnumerable<PackageReference>> GetInstalledPackagesAsync(CancellationToken token)
         {
-            PackageReference[] installedPackages;
+            var lockFilePath = await GetAssetsFilePathAsync();
```

Comment:
Only make this check whne necessary. 

That is, fetch the packageSpec first and then after the null check, get the assets file path if necessary.